docs/DeveloperGuide.md

```diff
@@ -1,22 +1,241 @@
+   backend logic. List of all `Messages`:
+
+`WELCOME`: Greeting message displayed at startup.
+
+`EXIT`: Farewell message when exiting.
+
+`ALLOCATE_COMPLETE`: Shown upon completion of the allocation process.
+
+`HELP`: Multi-line help message listing all available commands and their descriptions.
+
+`ERROR`: General error message for unexpected issues.
+
+`REVERT_COMPLETE`: Message displayed after a successful revert operation.
+
+Each message can be accessed and printed via `Messages.<MESSAGE_NAME>` in the `UI` class or any other class that 
+references it.
+
+#### Customizing and Extending the UI
+   Adding New Messages:
+1.  Open the `Messages` enum.
+2.  Add a new constant with the message text. Example:
+```java
+NEW_MESSAGE("Your custom message text here");
+```
+3. Reference the new message in the `UI` class or any other relevant part of the application using `Messages.NEW_MESSAGE`
+
+
+
+### Parser
+
+The `Parser` class is a crucial component instantiated as soon as FindOurSEP is initialized. Its responsibility is to process the user’s input into commands and invoking the correct command object for the rest of the program.
+
+Some of its core features include:
+- Breaking down user input and extracting the relevant command and data for further processing.
+- Provide robust error handling when unknown command is received.
+- Validate data parsed in from external file (.CSV, .JSON, .TXT) sources. (Further details in the `Storage` class)
+
+Here is a class diagram highlighting the fundamental structure of the `Parser` class.
+
+![ParserClass](UML_Diagrams/ParserClass.drawio.svg)
 
-## Design & implementation
+How `Parser` works:
+1. Whenever the user enters an input, the input will be directed to the `Parser` class's `parseInput()` method. 
+2. Within the method, the command will be extracted and the appropriate `XYZCommand` object (XYZCommand is a placeholder for various commands such as DeleteCommand, ListCommand, etc.) will be instantiated.
+3. Upon instantiation, the `XYZCommand` is prepared for execution. Each `XYZCommand` class, inheriting from the abstract `Command` class, has a `run()` method that executes its specific instructions.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The sequence diagram below demonstrates the interactions within the `Parser` component when a user inputs the command: `add id/A1234567I gpa/5.0 p/{13,61,43}`.
 
+![ParserSequence](UML_Diagrams/ParserSequence.drawio.svg)
+
+The boolean return value of `parseInput()` indicates whether the user has chosen to continue or terminate the program. A `true` value keeps FindOurSEP running, while a `false` value ends the program.
+
+### Commands
+
+#### Add Command
+
+#### Delete Command
+
+#### Find Command
+
+#### Filter Command
+
+#### Criteria Command
+
+#### Stats Command
+
+#### ViewQuota Command
+
+#### List Command
+
+#### Allocate Command
+
+#### Revert Command
+
```

Comment:
![image](https://github.com/user-attachments/assets/dd16fea9-c8bd-4041-834d-6075f0ecec0b)

Diagram image for RevertSequence and GenerateSequence is currently broken